provisioning/security/security-provider/src/main/java/com/microsoft/azure/sdk/iot/provisioning/security/SecurityProvider.java

```diff
@@ -131,34 +37,24 @@
     abstract public String getRegistrationId() throws SecurityProviderException;
 
     /**
-     * Retrieves the SSL context loaded with trusted certs. In case of X509 SSL context shall be loaded with complete chain
-     * all the way till the leaf along with its private key.
-     * @return The SSLContext relevant to the flow
-     * @throws SecurityProviderException If ssl context could not be generated for any of the reason
+     * Retrieves the SSL context loaded with trusted certs. This default implementation loads the trusted certificates
+     * from your device's trusted root certification authorities certificate store. Implementations of {@link SecurityProviderX509}
+     * must override this function so that it returns an SSLContext instance with the required private key and public certificates
+     * loaded into it as well as the default trusted certificates saved in your device's trusted root certification authorities certificate store.
+     * @return The SSLContext instance.
+     * @throws SecurityProviderException If ssl context could not be generated for any of the reason.
      */
-    abstract public SSLContext getSSLContext() throws SecurityProviderException;
-
-    KeyStore getKeyStoreWithTrustedCerts() throws NoSuchAlgorithmException, IOException, CertificateException, KeyStoreException
+    public SSLContext getSSLContext() throws SecurityProviderException
```

Comment:
What are the casing rules for Java for acronyms? Should this be `getSslContext`?

Reply:
Java is pretty mixed on this front. Since the type itself is SSLContext, it makes sense to me to keep this as getSSLContext rather than getSslContext